vividus/src/main/java/org/vividus/steps/SpringStepFactoryBeanPostProcessor.java

```diff
@@ -35,7 +38,13 @@ public Object postProcessAfterInitialization(Object bean, String beanName)
         if (bean instanceof SpringStepFactory)
         {
             List<String> stepBeanNames = BeanFactoryUtils.mergeLists(applicationContext, "stepBeanNames-");
-
+            List<String> duplicateStepBeanNames = CollectionUtils.getCardinalityMap(stepBeanNames).entrySet()
+                    .stream()
+                    .filter(stepBeanNameCount -> stepBeanNameCount.getValue() > 1)
+                    .map(Map.Entry::getKey)
+                    .collect(toList());
+            Validate.validState(
+                    duplicateStepBeanNames.isEmpty(), "Duplicate step beans name found %s", duplicateStepBeanNames);
```

Comment:
```suggestion
            String duplicateStepBeanNames = CollectionUtils.getCardinalityMap(stepBeanNames).entrySet()
                    .stream()
                    .filter(stepBeanNameCount -> stepBeanNameCount.getValue() > 1)
                    .map(Map.Entry::getKey)
                    .collect(Collectors.joining(", "));
            Validate.validState(
                    duplicateStepBeanNames.isEmpty(), "Duplicate step beans names are found: %s. Please, consider renaming to avoid conflicts", duplicateStepBeanNames);
```